opennms-model/src/main/java/org/opennms/netmgt/model/OnmsNode.java

```diff
@@ -448,14 +449,51 @@ public void setSysName(String nodesysname) {
         m_sysName = nodesysname;
     }
 
-    @Column(name="hasFlows", nullable=false)
-    @XmlAttribute(name="hasFlows")
+    @Transient
     public boolean getHasFlows() {
-        return m_hasFlows;
+        if (OnmsSnmpInterface.INGRESS_AND_EGRESS_REQUIRED) {
+            return getHasIngressFlows() && getHasEgressFlows();
+        } else {
+            return getHasIngressFlows() || getHasEgressFlows();
+        }
+    }
+
+    @Transient
+    public boolean getHasIngressFlows() {
+        if (m_lastIngressFlow == null) {
+            return false;
+        }
+        return (new Date().getTime() - m_lastIngressFlow.getTime()) / 1000 < OnmsSnmpInterface.MAX_FLOW_AGE;
```

Comment:
`new Date().getTime()` can be replaced by `System.currentTimeMillis()`

Reply:
done